code/modules/clothing/suits/_suits.dm

```diff
@@ -13,7 +13,13 @@
 	var/mutantrace_variation = NO_MUTANTRACE_VARIATION
 	var/adjusted = NORMAL_STYLE
 	limb_integrity = 0 // disabled for most exo-suits
+	var/obj/item/badge/attached_badge
+	var/mutable_appearance/badge_overlay
 
+/obj/item/clothing/suit/Destroy()
+	if(attached_badge)
+		QDEL_NULL(attached_badge)
+	return ..()
```

Comment:
attachable component when

Reply:
Never

Reply:
exactly

Reply:
This really should be a component.
I think TGMC has one if you want to look there for inspiration.